rio/src/RIO/Prelude/Reexports.hs

```diff
@@ -50,6 +50,8 @@ module RIO.Prelude.Reexports
   , Control.Monad.Reader.MonadReader
   , Control.Monad.Reader.MonadTrans(..)
   , Control.Monad.Reader.ReaderT(..)
+  , Control.Monad.State.MonadState(..)
```

Comment:
Since the method names here can conflict with other libraries in some cases, I'd rather not include them in the `RIO` module itself. We could add a `RIO.State`/`RIO.Writer` pair of modules, though.

Reply:
Wondering if it is worth adding a `Monad` in the name? I'm not sure, I guess we are almost certain there is never going to be something called `State` or `Writer` that doesn't imply the monads

Reply:
@snoyberg So if I understand correctly,  you would like to have a new module RIO.State & RIO.Writer (next to the RIO now which contains everything) that only contains the State and Writer monad stuff created in this PR? So that would mean to exclude the SomeRef, HasStateRef, MonadState from the main RIO module and only import these in two new modules (RIO.State and RIO.Writer)?

Reply:
I meant only the `MonadState (..)` and `MonadWriter (..)` reexports, which will cause name collisions, which we want to avoid in a prelude-like module. There's no problem with everything else that's been exported.

Reply:
I think I still don't get it, but I've added a commit to make it more visible. If i remove the MonadState deps from the Reexports, then I certainly need to include them in the RIO.Prelude.RIO file, otherwise MonadState / MonadWriter is not found. Now, I've also created the RIO.State and RIO.Writer which simply exposes the Control.Monad.State.MonadState and Control.Monad.Writer.MonadWriter. Does this makes sense? The changes can be seen in the commit.

Reply:
Yes, I think that addresses it. It avoids an import of the `RIO` module from including the `get` identifier, for example, which conflicts with some other libraries.

Reply:
Unfortunately, for some reason, that change is not appearing in the diff for this PR, and I can't understand why.